spec/controllers/admin/event_schedules_controller_spec.rb

```diff
@@ -0,0 +1,143 @@
+require 'spec_helper'
+
+describe Admin::EventSchedulesController do
+  let(:venue) { create(:venue) }
+  let(:conference) { create(:conference, venue: venue) }
+  let(:schedule) { create(:schedule, program: conference.program)}
+  let(:event_schedule) { create(:event_schedule, schedule: schedule)}
+  let!(:organizer_role) { Role.find_by(name: 'organizer', resource: conference) }
+  let(:organizer) { create(:user, role_ids: organizer_role.id) }
+
+  context 'logged in as an organizer' do
+    before :each do
+      sign_in(organizer)
+      event_schedule
+    end
+
+    describe 'POST #create' do
+      context 'with valid attributes' do
+        it 'saves the event schedule to the database' do
+          expected = expect do
+            post :create, conference_id: conference.short_title, event_schedule:
+                 attributes_for(:event_schedule,
+                                schedule_id: schedule.id,
+                                event_id: create(:event, program: conference.program).id,
+                                room_id: create(:room, venue: venue).id,
+                                start_time: conference.start_date)
+          end
+          expected.to change { EventSchedule.count }.by 1
+        end
+
+        it 'renders JSON without errors' do
+          post :create, conference_id: conference.short_title, event_schedule:
+               attributes_for(:event_schedule,
+                              schedule_id: schedule.id,
+                              event_id: create(:event, program: conference.program).id,
+                              room_id: create(:room, venue: venue).id,
+                              start_time: conference.start_date)
+
+          expect(response).to be_success
+          expect(JSON.parse(response.body)['status']).to eq('ok')
+        end
+      end
+
+      context 'with invalid attributes' do
+        it 'does not save the event schedule to the database' do
+          expected = expect do
+            post :create, conference_id: conference.short_title, event_schedule:
+                 attributes_for(:event_schedule,
+                                schedule_id: schedule.id,
+                                event_id: nil,
+                                room_id: nil,
+                                start_time: nil)
+          end
+          expected.to_not change { EventSchedule.count }
+        end
+
+        it 'renders JSON with error' do
+          post :create, conference_id: conference.short_title, event_schedule:
+               attributes_for(:event_schedule,
+                              schedule_id: schedule.id,
+                              event_id: nil,
+                              room_id: nil,
+                              start_time: nil)
+
+          expect(JSON.parse(response.body)['status']).to eq("The event couldn't be scheduled. Event can't be blank. Room can't be blank. Start time can't be blank")
```

Comment:
Just test the status code.

 Event can't be blank. Room can't be blank. Start time can't be blank etc are validation tests which we already cover in event_schedules_spec.rb